scripts/tidy.sh

```diff
@@ -2,7 +2,25 @@
 
 set -euo pipefail
 
-for f in $(git ls-files | grep go.mod)
+ROOT_DIR=$(git rev-parse --show-toplevel)
+SDK_DIR="$ROOT_DIR/sdk"
+PKG_DIR="$ROOT_DIR/pkg"
+
+for f in $(git ls-files | grep 'go\.mod')
 do
-    (cd "$(dirname "${f}")" && go mod tidy -compat=1.18)
+    DIR="$(dirname "${f}")"
+    pushd "${DIR}" >/dev/null
+    case "${DIR}" in
+        pkg|sdk|developer-docs)
+            echo "tidying '${DIR}'"
+            go mod tidy -compat=1.18
+            ;;
+        *)
+            echo "updating go mod replacements and tidying '${DIR}'"
+            go mod edit -replace "github.com/pulumi/pulumi/sdk/v3=$(realpath --relative-to=. "${SDK_DIR}")"
+            go mod edit -replace "github.com/pulumi/pulumi/pkg/v3=$(realpath --relative-to=. "${PKG_DIR}")"
```

Comment:
realpath does not support that argument on macOS.

```
% realpath --relative-to=. $(pwd)/foo
realpath: illegal option -- -
usage: realpath [-q] [path ...]
```

This might need to become a messier shell script or a Python program
if it needs to be smarter.

Reply:
In the past, rather than add that level of complexity and reinventing the wheel, I've made scripts error out if they detect macOS's 17 year old antiquated userland.

It's, IMO, much more rational than reinventing GNU coreutils.

Reply:
While I agree in principle, I think the problem here is that this script is meant to be run locally. CI just validates that you ran it. Folks on macOS will be unable to run this script. 

An alternative is to detect if 'grealpath' exists and prefer that. If it doesn't exist and 'realpath' fails, print an error suggesting a `brew install coreutils`.